src/components/results/property/Property.js

```diff
@@ -42,23 +50,76 @@ class Property extends React.Component {
     }
   }
 
+  onRent = async () => {
+    const { login, displayError } = this.props
+    const { activeUser } = this.context
+    if ( activeUser ) {
+      if (!this.state.canRent) {
+        displayError(new Error('HTTPS is required to use 2FA.'))
+        return;
+      } else if (!this.props.enrolled) {
+        displayError(new Error('No 2FA enrolled 2FA: Please enroll in 2FA (under Login/Profile menu at the top right) to Rent.'))
+        return;
+      }
+      this.setState({ loading: true })
+      try {
+        const accountName = await activeUser.getAccountName()
+        const rentChallenge = await generateRentChallenge(accountName, "aproperty")
+        const userAuth = await signRentChallenge(accountName, "aproperty", rentChallenge)
+        const transaction = generateRentTransaction(accountName, "aproperty", rentChallenge.serverKey, rentChallenge.userKey, rentChallenge.serverAuth, userAuth)
+        // The activeUser.signTransaction will propose the passed in transaction to the logged in Authenticator
+        await activeUser.signTransaction(transaction, transactionConfig)
+        this.setState({rented: true})
+      } catch (err) {
+        displayError(err)
+      }
+      this.setState({ loading: false })
+    } else {
+      login()
+    }
+  }
+
   render() {
-    const { loading, liked } = this.state
+    const { loading, liked, rented } = this.state
 
     return (
       <div className='property-container'>
         <div
           className='property-image'
           tabIndex={0}
           role='button'
-          onClick={this.onLike}
-          onKeyUp={event => onKeyUpEnter(event, this.onLike)}
+          aria-label='Rent Property Image'
+          onClick={this.onRent}
+          onKeyUp={event => onKeyUpEnter(event, this.onRent)}
         >
           <PropertyImage loading={loading} liked={liked} />
```

Comment:
I would think that the PropertyImage component should have a prop for the onClick for the component rather than this wrapping div